### PR TITLE
[runtime-transaction] add secp256r1 to precompile signature details

### PR DESCRIPTION
**Pull request by samkim-crypto**
#### Problem
The secp256r1 precompile (https://github.com/anza-xyz/agave/pull/3152) was added, but the logic in some of the crates in the monorepo such as `runtime-transaction` does not yet account for the new precompile.

#### Summary of Changes
Added a secp256r1 signatures field to the main types in the `runtime-transaction`.

The actual costs for the new precompile has to be added to the `cost-model` crate, but I will do it on a follow-up.

Fixes #

### CODE REVIEW

**Review by LucasSte — DISMISSED** [review]

File: runtime-transaction/src/signature_details.rs [hunk 1]
```diff
@@ -140,26 +150,31 @@ mod tests {
             Pubkey::new_unique(),
             solana_sdk::secp256k1_program::ID,
             solana_sdk::ed25519_program::ID,
+            solana_secp256r1_program::ID,
```
Comment (buffalojoec) [inline runtime-transaction/src/signature_details.rs]: Ahh bummer, is this ID not included in the SDK anywhere? Would avoid the dependency.

Comment (buffalojoec) [reply runtime-transaction/src/signature_details.rs]: Or maybe `solana-sdk-ids`?

**Review by buffalojoec — COMMENTED** [review]
Looks good, just curious if we can avoid the dependency or not.

File: runtime-transaction/src/signature_details.rs (hunk 1 above)
Comment (pgarg66) [reply runtime-transaction/src/signature_details.rs]: Maybe worth adding the ID to sdk in this PR.

Comment (samkim-crypto) [reply runtime-transaction/src/signature_details.rs]: Oh sorry I missed this. I replace all `solana_secp256r1_program` with `solana_sdk_ids::secp256r1_program`.

**Comment by samkim-crypto** [comment]
Updated! and the CI seems to pass!

**Review by buffalojoec — APPROVED** [review]

**Pull request merged**